components/browser/state/src/main/java/mozilla/components/browser/state/reducer/TopSiteReducer.kt

```diff
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.reducer
+
+import mozilla.components.browser.state.action.TopSiteAction
+import mozilla.components.browser.state.state.BrowserState
+
+internal object TopSiteReducer {
+    fun reduce(state: BrowserState, action: TopSiteAction): BrowserState {
+        return when (action) {
+            is TopSiteAction.AddTopSiteAction -> {
+                state.copy(
+                    topSites = state.topSites + action.topSite
+                )
+            }
+            is TopSiteAction.AddTopSitesAction -> {
+                state.copy(
+                    topSites = state.topSites + action.topSites
+                )
+            }
+            is TopSiteAction.RemoveTopSiteAction -> {
+                state.copy(
+                    topSites = state.topSites - action.topSite
+                )
+            }
+            is TopSiteAction.UpdateTopSiteAction -> {
+                val topSites = state.topSites
+                val topSiteIndex = topSites.indexOfFirst { it == action.topSite }
```

Comment:
I don't think this will work because of https://github.com/mozilla-mobile/android-components/pull/11490/files#r779303804

Reply:
Yeah, an ID would be helpful here. Even if it is only a transient UUID that never gets saved and is just unique for the runtime.